step/step.go

```diff
@@ -0,0 +1,255 @@
+package step
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/bitrise-io/go-steputils/v2/export"
+	"github.com/bitrise-io/go-steputils/v2/stepconf"
+	"github.com/bitrise-io/go-utils/v2/command"
+	"github.com/bitrise-io/go-utils/v2/env"
+	"github.com/bitrise-io/go-utils/v2/log"
+	"github.com/bitrise-io/go-utils/v2/pathutil"
+	"github.com/mattn/go-zglob"
+)
+
+const (
+	htmlReportDirKey = "BITRISE_HTML_REPORT_DIR"
+)
+
+type Input struct {
+	TestDeployDir    string `env:"test_result_dir,required"`
+	XcresultPatterns string `env:"xcresult_patterns"`
+	Verbose          bool   `env:"verbose,opt[true,false]"`
+}
+
+type Config struct {
+	TestDeployDir    string
+	XcresultPatterns []string
+}
+
+type Result struct {
+	HtmlReportDir string
+}
+
+type ReportGenerator struct {
+	envRepository  env.Repository
+	inputParser    stepconf.InputParser
+	commandFactory command.Factory
+	exporter       export.Exporter
+	logger         log.Logger
+}
+
+func NewReportGenerator(envRepository env.Repository, inputParser stepconf.InputParser, commandFactory command.Factory, exporter export.Exporter, logger log.Logger) ReportGenerator {
+	return ReportGenerator{
+		envRepository:  envRepository,
+		inputParser:    inputParser,
+		commandFactory: commandFactory,
+		exporter:       exporter,
+		logger:         logger,
+	}
+}
+
+func (r *ReportGenerator) ProcessConfig() (*Config, error) {
+	var input Input
+	err := r.inputParser.Parse(&input)
+	if err != nil {
+		return &Config{}, err
+	}
+
+	stepconf.Print(input)
+	r.logger.EnableDebugLog(input.Verbose)
+
+	patterns := strings.Split(strings.TrimSpace(input.XcresultPatterns), "\n")
+	var filteredPatterns []string
+
+	for _, p := range patterns {
+		pattern := strings.TrimSpace(p)
+		if pattern == "" {
+			continue
+		}
+
+		if !strings.HasSuffix(pattern, ".xcresult") {
+			return nil, fmt.Errorf("pattern (%s) must filter for xcresult files", pattern)
+		}
+
+		filteredPatterns = append(filteredPatterns, pattern)
+	}
+
+	return &Config{
+		TestDeployDir:    input.TestDeployDir,
+		XcresultPatterns: filteredPatterns,
+	}, nil
+}
+
+func (r *ReportGenerator) InstallDependencies() error {
+	r.logger.Println()
+	r.logger.Infof("Installing XCTestHTMLReport")
+
+	params := []string{"install", "bitrise-io/XCTestHTMLReport@speed-improvements", "--no-link"}
+	cmd := r.commandFactory.Create("mint", params, nil)
```

Comment:
Do we assume that mint is already installed? It could make sense to add it as a brew dependency if it is already installed from brew

Reply:
We will be removing mint completely once we have pre-built binaries. And we will do this before the release so this piece of code will be deleted.